port/unix/omrsysinfo.c

```diff
@@ -7557,3 +7559,81 @@ omrsysinfo_get_number_context_switches(struct OMRPortLibrary *portLibrary, uint6
 	return OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED;
 #endif /* defined(LINUX) */
 }
+
+/**
+ * Get the process ID and commandline for each process.
+ * @param[in] portLibrary The port library.
+ * @param[in] callback The function to be invoked for each process with the process ID and command info.
+ * @param[in] userData Data passed to the callback.
+ * @return 0 on success, or the first non-zero value returned by the callback.
+ */
+uintptr_t
+omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+	DIR *dir = NULL;
+	struct dirent *entry = NULL;
+	char cmdline_path[PATH_MAX];
+	char comm_path[PATH_MAX];
```

Comment:
Suggest we only need one of these buffers at a time (write the first path, try to read that file and only write the second path if necessary).